server/src/main/java/io/deephaven/server/barrage/BarrageStreamGenerator.java

```diff
@@ -810,36 +810,31 @@ private ByteBuffer getSnapshotMetadata(final SnapshotView view) throws IOExcepti
         }
 
         int effectiveColumnSetOffset = 0;
-        if (isSnapshot && view.subscribedColumns != null) {
+        if (view.subscribedColumns != null) {
             effectiveColumnSetOffset = new BitSetGenerator(view.subscribedColumns).addToFlatBuffer(metadata);
         }
 
         final int rowsAddedOffset = rowsAdded.addToFlatBuffer(metadata);
-        final int rowsRemovedOffset = 0;
+        final int addedRowsIncludedOffset = rowsIncluded.addToFlatBuffer(metadata);
```

Comment:
Relative to `getSubscriptionMetadata`, we are adding `rowsIncluded` out-of-order w.r.t. `shifted`. Is this a bug?
Related, do we need to do this if `!view.isViewport()`?

Reply:
out-of-order: I believe this is correct (or at least permissible) - the data just needs to have been put in the flatbuffer before we first reference it, the actual order of the data doesn't matter.

Reply:
Made changes to `getSnapshotMetadata()` to re-order shifts / rowIncluded to match exactly the `barrage.fbs` definition.  Also re-ordered `BarrageUpdateMetadata.addEffectiveReverseViewport()` calls.  Now we are consistent with the definition.